test/hotspot/jtreg/runtime/ErrorHandling/BadNativeStackInErrorHandlingTest.java

```diff
@@ -58,44 +58,15 @@ public static void main(String[] args) throws Exception {
     output_detail.shouldMatch("# +(?:SIGSEGV|SIGBUS|EXCEPTION_ACCESS_VIOLATION).*");
 
     // extract hs-err file
-    String hs_err_file = output_detail.firstMatch("# *(\\S*hs_err_pid\\d+\\.log)", 1);
-    if (hs_err_file == null) {
-        throw new RuntimeException("Did not find hs-err file in output.\n");
-    }
-
-    File f = new File(hs_err_file);
-    if (!f.exists()) {
-        throw new RuntimeException("hs-err file missing at " +
-                                   f.getAbsolutePath() + ".\n");
-    }
-
-    System.out.println("Found hs_err file. Scanning...");
-
-    FileInputStream fis = new FileInputStream(f);
-    BufferedReader br = new BufferedReader(new InputStreamReader(fis));
-    String line = null;
+    File hs_err_file = HsErrFileUtils.openHsErrFileFromOutput(output_detail);
 
     // The failing line looks like this:
     // [error occurred during error reporting (printing native stack), id 0xb]
-    Pattern pattern =
-        Pattern.compile("\\[error occurred during error reporting \\(printing native stack\\), id .*\\]");
-
-    String lastLine = null;
-    while ((line = br.readLine()) != null) {
-        if (pattern.matcher(line).matches()) {
-            System.out.println("Found: " + line + ".");
-            throw new RuntimeException("hs-err file should not contain: '" +
-                                       pattern + "'");
-        }
-        lastLine = line;
-    }
-    br.close();
+    Pattern negativePatterns[] = {
+        Pattern.compile("\\[error occurred during error reporting \\(printing native stack\\), id .*\\]")
+    };
 
-    if (!lastLine.equals("END.")) {
-        throw new RuntimeException("hs-err file incomplete (missing END marker.)");
-    } else {
-        System.out.println("End marker found.");
-    }
+    HsErrFileUtils.checkHsErrFileContent(hs_err_file, null, negativePatterns, true, false);
```

Comment:
Can we annotate the true/false parameters so the reader doesn't have to go and look up what they mean i.e. `true /* check END */, false /*not verbose */` ?

Reply:
My IDE usually does this for me, but I can make it explicit, sure.